signer/src/transaction_signer.rs

```diff
@@ -1218,54 +1383,83 @@ where
+            Some(state_machine) => state_machine.process(msg).map_err(Error::Wsts)?,
+            None => {
+                tracing::warn!("missing signing round");
+                return Err(Error::MissingStateMachine(*state_machine_id));
+            }
         };
 
+        // Check and store if this is a DKG verification-related message.
+        let is_dkg_verification = matches!(
+            state_machine_id,
+            StateMachineId::DkgVerification(_, chain_tip) if chain_tip.block_hash == *bitcoin_chain_tip
+        );
+
         // If this is a DKG verification then we need to process the message in
         // the frost coordinator as well to be able to properly follow the
         // signing round (which is otherwise handled by the signer state
-        // machine).
-        let mut frost_coordinator = if let StateMachineId::RotateKey(_, _) = state_machine_id {
-            self.dkg_verification_state_machines
-                .get_mut(&state_machine_id)
-        } else {
-            None
-        };
-
-        let outbound_messages = state_machine.process(msg).map_err(Error::Wsts)?;
+        // machine). We pass `None` as the `signer_id` because we have just
+        // validated the sender above.
+        if is_dkg_verification {
+            self.process_dkg_verification_message(*state_machine_id, sender, None, msg)
+                .await?;
+        }
 
+        // The WSTS state machines assume we read our own messages, so if the
+        // state machine emitted any outbound messages then we need to process
+        // them manually. We ignore any extra messages emitted from these calls.
         for outbound_message in outbound_messages.iter() {
-            // The WSTS state machine assumes we read our own messages
-            state_machine
+            // Process in the signer state machine.
+            self.wsts_state_machines
+                .get_mut(state_machine_id)
+                .ok_or_else(|| Error::MissingStateMachine(*state_machine_id))?
                 .process(outbound_message)
                 .map_err(Error::Wsts)?;
 
-            // Process the message in the frost coordinator as well, if we have
-            // one. Note that we _do not_ send any messages to the network; the
-            // frost coordinator is only following the round.
-            if let Some(ref mut frost_coordinator) = frost_coordinator {
-                frost_coordinator.process_message(outbound_message)?;
+            // If this is a DKG verification then we need to process the message
+            // in the FROST state machine as well for it to properly follow
+            // the signing round.
+            if is_dkg_verification {
+                self.process_dkg_verification_message(
+                    *state_machine_id,
+                    sender,
+                    signer_id,
+                    outbound_message,
+                )
+                .await?;
```

Comment:
Ahh right. We do not receive our own messages in this case because the FROST state machine lives in the `TxSignerEventLoop` and not in `TxCoordinatorEventLoop`.